components/request-card/script.js

```diff
@@ -108,6 +108,273 @@ async function createRequestCardComponent({
+    taskDetailsContainer.appendChild(taskStatusContainer);
+  }
+  taskStatusContainer.appendChild(taskStatusTextLabel);
+
+  taskStatusContainer.appendChild(taskStatusElement);
+  summaryContainer.appendChild(datesContainer);
+
+  newDeadlineValue.appendChild(newEndsOnTooltip);
+
+  requestForValue.appendChild(requestCreatedAtTooltip);
+
+  const requestNumberContainer = createElement({ type: 'div' });
+  if (isExtensionRequest) {
+    datesContainer.appendChild(requestNumberContainer);
+  } else if (data.type !== RequestType.OOO) {
+    taskDetailsContainer.appendChild(requestNumberContainer);
+  }
+  requestNumberContainer.appendChild(requestNumberLabel);
+
+  requestNumberContainer.appendChild(requestNumberValue);
+  cardAssigneeButtonContainer.appendChild(assigneeContainer);
+  assigneeContainer.appendChild(assigneeTextLabel);
+  assigneeContainer.appendChild(assigneeImage);
+
+  formContainer.appendChild(summaryContainer);
+
+  assigneeContainer.appendChild(assigneeNameElement);
+  cardAssigneeButtonContainer.appendChild(requestActionContainer);
+
+  panel.appendChild(reasonContainer);
+
+  if (isExtensionRequest) {
+    panel.appendChild(logContainer);
+  } else if (data?.state !== RequestStatus.PENDING) {
+    panel.appendChild(commentContainer);
+  }
+
+  accordionButton.addEventListener('click', function () {
+    isExtensionRequest ? renderLogs(data.id) : toggleAccordionPanel(panel);
+  });
+  const cardFooter = createElement({ type: 'div' });
+  cardFooter.appendChild(cardAssigneeButtonContainer);
+  cardFooter.appendChild(accordionContainer);
+  formContainer.appendChild(cardFooter);
+  rootElement.appendChild(formContainer);
+
+  formContainer.addEventListener('submit', async (e) => {
+    e.preventDefault();
+    const formData = getFormEntries(new FormData(e.target));
+    formData['newEndsOn'] = new Date(formData['newEndsOn']).getTime() / 1000;
+
+    const todayDate = Math.floor(new Date().getTime() / 1000);
+    if (
+      !formData.title?.trim() ||
+      !formData.reason?.trim() ||
+      isNaN(formData['newEndsOn']) ||
+      formData['newEndsOn'] < todayDate
+    ) {
+      return;
+    }
+
+    const removeSpinner = addLoadingSpinner(rootElement);
+    rootElement.classList.add('disabled');
+    const revertDataChange = updateCardData(formData);
+    const payloadForLog = {
+      body: {
+        ...(formData?.newEndsOn !== requestDetails.newEndsOnInMillisecond && {
+          newEndsOn: formData.newEndsOn,
+          oldEndsOn: requestDetails.oldEndsOnInMillisecond,
+        }),
+        ...(formData?.reason !== data.reason && {
+          newReason: formData.reason,
+          oldReason: data.reason,
+        }),
+        ...(formData?.title !== data.title && {
+          newTitle: formData.title,
+          oldTitle: data.title,
+        }),
+      },
+      meta: {
+        requestId: data.id,
+        name: `${currentUser?.first_name ?? ''} ${
+          currentUser?.last_name ?? ''
+        }`,
```

Comment:
```suggestion
  name: [currentUser?.first_name, currentUser?.last_name].filter(Boolean).join(' '),
```

can we try this way?